From b1c11c7bf4ece7059e80c8912f7c7ff84a868761 Mon Sep 17 00:00:00 2001
From: "Badrappan, Jeevaka" <jeevaka.badrappan@intel.com>
Date: Tue, 4 Jul 2023 22:45:40 +0530
Subject: [PATCH] Do firmware update of graphics and wifi inside ubu_update_fw

Changes include:
- Move the firmware update of graphics and wifi inside ubu_update_fw
to avoid duplication in multiple functions.
- Update firmware tag to 20230625
- Remove not used wifi firmware update function.

Tracked-On: OAM-111023
Signed-off-by: Badrappan, Jeevaka <jeevaka.badrappan@intel.com>
---
 host_scripts/setup_host.sh | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/host_scripts/setup_host.sh b/host_scripts/setup_host.sh
index 7a55892..6816d45 100755
--- a/host_scripts/setup_host.sh
+++ b/host_scripts/setup_host.sh
@@ -204,14 +204,15 @@ function ubu_enable_host_sriov(){
 }
 
 function ubu_update_fw(){
-    FW_REL="linux-firmware-20221109"
+    FW_REL="linux-firmware-20230625"
 
-    [ ! -f $CIV_WORK_DIR/$FW_REL.tar.xz ] && wget "https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/snapshot/linux-firmware-20221109.tar.gz" -P $CIV_WORK_DIR
+    [ ! -f $CIV_WORK_DIR/$FW_REL.tar.xz ] && wget "https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/snapshot/linux-firmware-20230625.tar.gz" -P $CIV_WORK_DIR
 
     [ -d $CIV_WORK_DIR/$FW_REL ] && rm -rf $CIV_WORK_DIR/$FW_REL
     tar -xf $CIV_WORK_DIR/$FW_REL.tar.gz
 
     cp -r $CIV_WORK_DIR/$FW_REL/i915/ /lib/firmware/
+    cp $CIV_WORK_DIR/$FW_REL/iwl* /lib/firmware/
 
     cd $CIV_WORK_DIR
     rm -rf $CIV_WORK_DIR/$FW_REL*
@@ -453,14 +454,6 @@ function ubu_update_bt_fw() {
     fi
 }
 
-function ubu_update_wifi_fw(){
-    if [ -d "linux-firmware" ] ; then
-            rm -rf linux-firmware
-    fi
-    git clone https://kernel.googlesource.com/pub/scm/linux/kernel/git/firmware/linux-firmware
-    sudo cp linux-firmware/iwlwifi-so-a0-hr-b0-64.ucode /lib/firmware
-}
-
 function setup_remote_infer() {
     echo "setting up remote inference"
     source $CIV_WORK_DIR/scripts/setup_ai_dispatcher.sh $1
-- 
2.40.0

